templates/ingress/ingress.yaml

```diff
@@ -1,29 +1,31 @@
 {{- if eq .Values.expose.type "ingress" }}
 {{- $ingress := .Values.expose.ingress -}}
 {{- $tls := .Values.expose.tls -}}
-{{- if eq .Values.expose.ingress.controller "gce" }}
-  {{- $_ := set . "portal_path" "/*" -}}
-  {{- $_ := set . "api_path" "/api/*" -}}
-  {{- $_ := set . "service_path" "/service/*" -}}
-  {{- $_ := set . "v2_path" "/v2/*" -}}
-  {{- $_ := set . "chartrepo_path" "/chartrepo/*" -}}
-  {{- $_ := set . "controller_path" "/c/*" -}}
-  {{- $_ := set . "notary_path" "/" -}}
-{{- else if eq .Values.expose.ingress.controller "ncp" }}
-  {{- $_ := set . "portal_path" "/.*" -}}
+{{- if semverCompare "<1.19-0" .Capabilities.KubeVersion.GitVersion }}
+{{- if eq .Values.expose.ingress.controller "ncp" }}
   {{- $_ := set . "api_path" "/api/.*" -}}
   {{- $_ := set . "service_path" "/service/.*" -}}
   {{- $_ := set . "v2_path" "/v2/.*" -}}
   {{- $_ := set . "chartrepo_path" "/chartrepo/.*" -}}
   {{- $_ := set . "controller_path" "/c/.*" -}}
+  {{- $_ := set . "portal_path" "/.*" -}}
   {{- $_ := set . "notary_path" "/.*" -}}
 {{- else }}
-  {{- $_ := set . "portal_path" "/" -}}
+  {{- $_ := set . "api_path" "/api/*" -}}
+  {{- $_ := set . "service_path" "/service/*" -}}
+  {{- $_ := set . "v2_path" "/v2/*" -}}
+  {{- $_ := set . "chartrepo_path" "/chartrepo/*" -}}
+  {{- $_ := set . "controller_path" "/c/*" -}}
+  {{- $_ := set . "portal_path" "/*" -}}
+  {{- $_ := set . "notary_path" "/*" -}}
+{{- end }}
+{{- else }}
```

Comment:
There are now two else-blocks here. Is one of them intended to read:
`{{- else if eq .Values.expose.ingress.controller "alb" }}` ?
This will not be evaluated by Helm correctly as is.

Reply:
Removed two else-blocks in the commit below.
If kubernetes version >= 1.19, path contains postfix "/" and pathType is "Prefix".
otherwise, path contains postfix "*" and pathType is "ImplementationSpecific".